core/templates/dev/head/galleries/Gallery.js

```diff
@@ -196,14 +219,22 @@ oppia.controller('Gallery', [
       }
     });
 
-    // SEARCH FUNCTIONALITY
+    // Transforms the category name of each exploration into a translation id
+    var addTranslationIds = function(allExplorationsInOrder) {
+      for (var i = 0; i < allExplorationsInOrder.length; i++) {
+        allExplorationsInOrder[i].i18nCategory = getI18nId('categories',
+          allExplorationsInOrder[i].category);
+      }
+    };
 
+    // SEARCH FUNCTIONALITY
     $scope.allExplorationsInOrder = [];
 
     // Called when the page loads, and after every search query.
     var _refreshGalleryData = function(data, hasPageFinishedLoading) {
       $scope.searchIsLoading = false;
       $scope.allExplorationsInOrder = data.explorations_list;
+      addTranslationIds($scope.allExplorationsInOrder);
```

Comment:
Slightly nicer, I think, if addTranslationIds returns something and we set $scope.allExplorationsInOrder to it. Otherwise this reads as though the translation ids are being added to some kind of other list somewhere.

It's hard to find a good name, too. Maybe populateTranslationIds? Not quite it, but it gives me more of a sense of 'filling something in' than 'add' does.